traveler-ios-ui/TravelerKitUI/ViewControllers/BookingItemSearch/BookingItemSearchSortFilterViewController.swift

```diff
@@ -0,0 +1,195 @@
+            let vc = navVC.topViewController as? BookingItemSearchFilterPriceViewController
+            vc?.delegate = self
+            vc?.currentRange = currentRange
+            vc?.fullRange = fullRange
+        default:
+            Log("Unknown segue", data: segue, level: .warning)
+            break
+        }
+    }
+    
+    @IBAction func cancel() {
+        dismiss(animated: true, completion: nil)
+    }
+    
+    @IBAction func submit() {
+        guard let currentSearchResult = currentSearchResult, searchQuery != nil else {
+            cancel()
+            return
+        }
+        
+        let sortData = currentSort!.data
+        var bookingItemSearchFilters = BookingItemSearchFilters()
+        bookingItemSearchFilters.priceRange = PriceRangeFilter(range: currentRange!, currency: currentSearchResult.facets.minPrice.baseCurrency)
+        bookingItemSearchFilters.categories = searchQuery!.categories
+        bookingItemSearchFilters.city = searchQuery!.city
+        bookingItemSearchFilters.country = searchQuery!.country
+        bookingItemSearchFilters.sortOption = sortData.sortOption
+        bookingItemSearchFilters.sortOrder = sortData.sortOrder
+        searchQuery = searchQuery!.filterSearchWith(bookingItemSearchFilters)
+        delegate?.bookingItemSearchSortFilterController(self, didSubmit: searchQuery)
+    }
+    
+    func resetData() {
+        currentSort = defaultSort
+        
+        if let minFacetsPrice = currentSearchResult?.facets.minPrice.valueInBaseCurrency, let maxFacetsPrice = currentSearchResult?.facets.maxPrice.valueInBaseCurrency, minFacetsPrice <= maxFacetsPrice {
+            currentRange = minFacetsPrice...maxFacetsPrice
+        } else {
+            currentRange = fullRange
+        }
+    }
+    
+    @IBAction func reset() {
+        resetData()
+        tableView.reloadData()
+    }
+    
+    public override func numberOfSections(in tableView: UITableView) -> Int {
+        return 2
+    }
+    
+    public override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        if section == 0 {
+            return 1
+        } else {
+            return 1
+        }
+    }
+    
+    public override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 45
+    }
+    
+    public override func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        if section == 0 {
+            return "Sort"
+        } else {
+            return "Filters"
+        }
+    }
+    
+    public override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellIdentifier, for: indexPath)
+        cell.selectionStyle = .none
+        
+        guard let searchSortFilterCell = cell as? BookingItemSearchSortFilterCell else {
```

Comment:
This guard statement is unnecessary. Set the reuse identifier for the cell in the storyboard and cast the cell as BookingItemSearchSortFilterCell when calling `dequeueReusableCell `.